### ISSUE TITLE
Errors with ansible 2.1

### ISSUE THREAD

**Issue by tonydoval** [cmprescott/ansible-xml#64]
I'' getting errors like:
With ansible 2.1
al: [desa-wl12c-1.edisa.com]: FAILED! => {"changed": false, "failed": true, "module_stderFile \"/tmp/ansible_wnslMv/ansible_module_xml.py\", line 424, in <module>\r\n    main()\r\n main\r\n    get_element_text(x, xpath, namespaces, module)\r\n  File \"/tmp/ansible_wnslMv/at is_node(tree, xpath, namespaces):\r\n  File \"/tmp/ansible_wnslMv/ansible_module_xml.py\",ces):\r\n  File \"/tmp/ansible_wnslMv/ansible_module_xml.py\", line 154, in xpath_matches\r\.etree.pyx\", line 2091, in lxml.etree._ElementTree.xpath (src/lxml/lxml.etree.c:53358)\r\n r.__init__ (src/lxml/lxml.etree.c:133440)\r\n  File \"xpath.pxi\", line 276, in lxml.etree.XFile \"xpath.pxi\", line 148, in lxml.etree._XPathEvaluatorBase.__init__ (src/lxml/lxml.etreContext.__init__ (src/lxml/lxml.etree.c:130211)\r\n  File \"extensions.pxi\", line 82, in lxalueError: need more than 1 value to unpack\r\n", "msg": "MODULE FAILURE", "parsed": false}

**Comment by mss** [comment]
Same here.

With version 2.0.0.2 (Ubuntu 16.04):

```
TASK [Set general localconfig options] *****************************************
ok: [zmadm1.example.com] => (item={'value': u'no', 'key': u'smtp_notify'})
```

With version 2.1.0.0 (PPA):

```
failed: [zmadm1.example.com] (item={'value': u'no', 'key': u'smtp_notify'}) => {"failed": true, "item": {"key": "smtp_notify", "value": "no"}, "module_stderr": "", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_M9m9a4/ansible_module_xml.py\", line 410, in <module>\r\n    main()\r\n  File \"/tmp/ansible_M9m9a4/ansible_module_xml.py\", line 406, in main\r\n    set_target(x, xpath, namespaces, attribute, value, module)\r\n  File \"/tmp/ansible_M9m9a4/ansible_module_xml.py\", line 223, in set_target\r\n    if not is_node(tree, xpath, namespaces):\r\n  File \"/tmp/ansible_M9m9a4/ansible_module_xml.py\", line 134, in is_node\r\n    if xpath_matches(tree, xpath, namespaces):\r\n  File \"/tmp/ansible_M9m9a4/ansible_module_xml.py\", line 154, in xpath_matches\r\n    if tree.xpath(xpath, namespaces=namespaces):\r\n  File \"lxml.etree.pyx\", line 2091, in lxml.etree._ElementTree.xpath (src/lxml/lxml.etree.c:53358)\r\n  File \"xpath.pxi\", line 342, in lxml.etree.XPathDocumentEvaluator.__init__ (src/lxml/lxml.etree.c:133440)\r\n  File \"xpath.pxi\", line 276, in lxml.etree.XPathElementEvaluator.__init__ (src/lxml/lxml.etree.c:132591)\r\n  File \"xpath.pxi\", line 148, in lxml.etree._XPathEvaluatorBase.__init__ (src/lxml/lxml.etree.c:131208)\r\n  File \"xpath.pxi\", line 57, in lxml.etree._XPathContext.__init__ (src/lxml/lxml.etree.c:130211)\r\n  File \"extensions.pxi\", line 82, in lxml.etree._BaseContext.__init__ (src/lxml/lxml.etree.c:121254)\r\nValueError: need more than 1 value to unpack\r\n", "msg": "MODULE FAILURE", "parsed": false}
```
The tasks:

```yaml
  - name: Ensure python-lxml is installed
    yum:
      name: "python-lxml"
      state: present
  - name: Register localconfig state
    stat:
      path: "/opt/zimbra/conf/localconfig.xml"
    register: zimbra_localconfig_stat
  - name: Set general localconfig options
    xml:
      file: "/opt/zimbra/conf/localconfig.xml"
      xpath: "/localconfig/key[@name='{{item.key}}']/value"
      value: "{{item.value}}"
    with_dict: "{{zimbra_localconfig}}"
    when:
      zimbra_localconfig_stat.stat.exists
```
And vars:
```
zimbra_localconfig:
  smtp_notify: "no"
```

There are actually more than a single var, I stripped it down. Also, it seems like it fails on *some* hosts and works on others which is especially puzzling.

**Comment by mss** [comment]
The patch from @h0nIg's branch which was 72864f at some point but got rebased by now fixes the issue for me.

@h0nIg maybe you could create an extra branch and pull request just for this issue?

**Comment by h0nIg** [comment]
just checkout https://github.com/cmprescott/ansible-xml/pull/65/commits/38b84e1f736c44e411ed947ae622a91752337c58

**Comment by mss** [comment]
I hope @cmprescott is more likely to merge your fix if it is submitted as a separate pull request instead of a `devel_misc` branch.

**Comment by cmprescott** [comment]
@mss - you read my mind. 

@h0nIg - Do you mind separating out the fix from the new features?

**Comment by mss** [comment]
I cherry picked the fix into #66.
